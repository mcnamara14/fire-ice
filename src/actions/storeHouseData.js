export const storeHouseData = (houseData) => ({
  type: 'STORE_HOUSE_DATA',
  houseData
});