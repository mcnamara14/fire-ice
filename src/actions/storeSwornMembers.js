export const storeSwornMembers = (houseName, swornMembers) => ({
  type: 'STORE_SWORN_MEMBERS',
  houseName,
  swornMembers
});