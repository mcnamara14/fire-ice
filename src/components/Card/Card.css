h2 {
  color: #fff;
  text-transform: uppercase;
  margin-bottom: 30px;
}

h3, h4 {
  color: #eee;
}

h3 {
  color: #841F28;
  font-size: 1.5rem;
  text-shadow: none;
}

h4 {
  font-size: 1.2rem;
}

p { 
  color: #fff;
}

.houseCard {
  width: calc(30% - 30px);
  background: #222;
  margin-bottom: 30px;
  padding: 20px;
  border-radius: 5px;
  border: 1px solid #ccc;
}
