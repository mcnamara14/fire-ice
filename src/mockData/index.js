export const mockCleanHouseData = [
  {
    ancestralWeapons: ["Lady Forlorn"],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    founded: "",
    name: "House Corbray of Heart's Home",
    seats: ["Heart's Home"],
    swornMembers: ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: ["Lord of Heart's Home", "King of the Fingers (historical)"],
    words: ""
  }
]

export const mockDirtyHouseData = [
  {
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words:""
  }
]