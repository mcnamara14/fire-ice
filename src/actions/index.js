import { storeHouseData } from './storeHouseData';
import { storeSwornMembers } from './storeSwornMembers';

export { 
  storeHouseData,
  storeSwornMembers
};