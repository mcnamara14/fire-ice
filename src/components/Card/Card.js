import React, { Component } from 'react';
import './Card.css';

export const Card = (props) => {
  const {
    name,
    founded, 
    seats,
    titles,
    coatOfArms,
    ancestralWeapons,
    words
  } = props;

  const foundedEra = founded ? founded : 'N/A';
  const allSeats = seats.map((seat, index) => {
    return (
      <p className="seat"  key={index}>seat: {seat}</p>
    );
  });

  return (
    <article className="houseCard">
      <h2>{name}</h2>
      <h3>{words}</h3>
      <h4>founded: {foundedEra}</h4>
      <div className="seats">
        {allSeats}
      </div>
    </article>
  ) 
}