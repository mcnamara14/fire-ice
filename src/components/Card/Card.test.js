import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps } from './Card';
import * as mockData from '../../mockData/index';
import { storeHouseData } from '../../actions/storeHouseData';
jest.mock('../../cleaners/cleanHouseData');
jest.mock('../../apiCalls/index');
 
describe('Card', () => {
  it('should match the snapshot', () => {
    const mockSeats = ['seat1'];
    const mockTitles = ['title1', 'title2'];
    const mockAncestralWeaopns = ['weapon1'];

    const wrapper = shallow(<Card seats={mockSeats} titles={mockTitles} ancestralWeapons={mockAncestralWeaopns} />);

    expect(wrapper).toMatchSnapshot();
  });
});