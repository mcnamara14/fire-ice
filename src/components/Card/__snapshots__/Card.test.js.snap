// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    ancestralWeapons={
        Array [
            "weapon1",
          ]
    }
    seats={
        Array [
            "seat1",
          ]
    }
    titles={
        Array [
            "title1",
            "title2",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <h3 />,
        <h4>
          Founded: 
          N/A
</h4>,
        <div
          className="seats"
>
          <p
                    className="seat"
          >
                    Seat: 
                    seat1
          </p>
</div>,
        <div
          className="titles"
>
          <p
                    className="title"
          >
                    Title: 
                    title1
          </p>
          <p
                    className="title"
          >
                    Title: 
                    title2
          </p>
</div>,
        <div
          className="weapons"
>
          <p
                    className="weapon"
          >
                    Weapon: 
                    weapon1
          </p>
</div>,
        <p>
          Coat of Arms: 
</p>,
        undefined,
      ],
      "className": "houseCard",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "N/A",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "N/A",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="seat"
>
              Seat: 
              seat1
</p>,
          ],
          "className": "seats",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Seat: ",
                "seat1",
              ],
              "className": "seat",
            },
            "ref": null,
            "rendered": Array [
              "Seat: ",
              "seat1",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="title"
>
              Title: 
              title1
</p>,
            <p
              className="title"
>
              Title: 
              title2
</p>,
          ],
          "className": "titles",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Title: ",
                "title1",
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              "Title: ",
              "title1",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Title: ",
                "title2",
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              "Title: ",
              "title2",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="weapon"
>
              Weapon: 
              weapon1
</p>,
          ],
          "className": "weapons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Weapon: ",
                "weapon1",
              ],
              "className": "weapon",
            },
            "ref": null,
            "rendered": Array [
              "Weapon: ",
              "weapon1",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: ",
          undefined,
        ],
        "type": "p",
      },
      undefined,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <h3 />,
          <h4>
            Founded: 
            N/A
</h4>,
          <div
            className="seats"
>
            <p
                        className="seat"
            >
                        Seat: 
                        seat1
            </p>
</div>,
          <div
            className="titles"
>
            <p
                        className="title"
            >
                        Title: 
                        title1
            </p>
            <p
                        className="title"
            >
                        Title: 
                        title2
            </p>
</div>,
          <div
            className="weapons"
>
            <p
                        className="weapon"
            >
                        Weapon: 
                        weapon1
            </p>
</div>,
          <p>
            Coat of Arms: 
</p>,
          undefined,
        ],
        "className": "houseCard",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "N/A",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "N/A",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="seat"
>
                Seat: 
                seat1
</p>,
            ],
            "className": "seats",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Seat: ",
                  "seat1",
                ],
                "className": "seat",
              },
              "ref": null,
              "rendered": Array [
                "Seat: ",
                "seat1",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="title"
>
                Title: 
                title1
</p>,
              <p
                className="title"
>
                Title: 
                title2
</p>,
            ],
            "className": "titles",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Title: ",
                  "title1",
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                "Title: ",
                "title1",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Title: ",
                  "title2",
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                "Title: ",
                "title2",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="weapon"
>
                Weapon: 
                weapon1
</p>,
            ],
            "className": "weapons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Weapon: ",
                  "weapon1",
                ],
                "className": "weapon",
              },
              "ref": null,
              "rendered": Array [
                "Weapon: ",
                "weapon1",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: ",
            undefined,
          ],
          "type": "p",
        },
        undefined,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
